468 mobile implementation of forum pages

anilanlar opened the pull request:
### Forum Feed Page
Implemented the Forum Feed Page. The color palette and the design is similar to that of Quiz Feed page.
 
![image](https://github.com/user-attachments/assets/e3f3985c-642c-4d3e-90f8-5db02cf9c00c)

### Forum Post Page
Implemented the Forum Post Page. It includes the comment section as well.

![image](https://github.com/user-attachments/assets/266ccd43-fb94-436d-942a-1cbb1f33bf36)

### Forum Post Creation Page

Implemented the Forum Post Creation Page. The color palette and the design is similar to that of Quiz Creation page.
![image](https://github.com/user-attachments/assets/c8fe6b75-f289-477d-9e74-ef0c77dfa75b)

YavizGuldalf commented on mobile/bulingo/app/forumPostPage.tsx:
```diff
@@ -0,0 +1,231 @@
+
+  return (
+    <ScrollView style={styles.container}>
+      <PostCard
```
Typescript throws a warning for this PostCard component, as it doesn't contain the `onPress` and `tags` properties. If those props are supposed to be optional, consider making them so in the PostCardProps type definition.

```typescript
type PostCardProps = {
  ...
  onPress?: ()=>void,
  tags?: string[],
  ...
}
```

YavizGuldalf commented on mobile/bulingo/app/forumPostPage.tsx:
```diff
@@ -0,0 +1,231 @@
+      {loading ? (
+        <ActivityIndicator size="large" color="#0000ff" />
+      ) : (
+        <FlatList
```
This usage (a FlatList inside a ScrollView) causes expo to throw a warning. The warning message is as follow: 


"VirtualizedLists should never be nested inside plain ScrollViews with the same orientation because it can break windowing and other functionality - use another VirtualizedList-backed container instead."

In order to not run into this issue, consider removing the ScrollView encapsulating the FlatList, and moving everything that was above the FlatList in the original ScrollView into the `listHeaderComponent` prop of the FlatList. You can see an example of this implementation at in the [profile page](https://github.com/bounswe/bounswe2024group6/blob/76063a3c676c51e6905567e0a6e03110eca07ac0/mobile/bulingo/app/(tabs)/profile.tsx#L95).

YavizGuldalf commented on mobile/bulingo/app/components/commentcard.tsx:
```diff
@@ -0,0 +1,120 @@
+const styles = StyleSheet.create({
+    cardContainer: {
+        flexDirection: 'row',
+        alignItems: 'center',
```
I think having the text centered in the commentCard looks a little bit off is the comment is short. Having it start from the top would look better in my opinion.

YavizGuldalf reviewed (changes requested):
When I tried to checkout this branch and run it with `npm install` followed by `npm start`, I ran into an error. It's about the import of `./api/forum`. There is no such file in this PR, did you forget to include it by any chance?

Also, the like button is missing in the 'CommentCard' component.

anilanlar commented:
Thank you for the feedback!

- I forgot adding forum.tsx file, I added it and now it should be resolved.
- I removed ScrollView component from forumPostPage.tsx
- I have modified the PostCard component, now it accepts the onPress and tags as parameters.
- I have added the missing like functionality to CommentCard component.

Could you go over my PR again?

YavizGuldalf commented on mobile/bulingo/app/forumPostPage.tsx:
```diff
@@ -0,0 +1,284 @@
+import PostCard from './components/postcard'; // Assuming you already have the PostCard component defined
+import CommentCard from './components/commentcard'; // Assuming you already have the CommentCard component defined
+import { useLocalSearchParams } from 'expo-router';
+import { fetchForumPostWithId, fetchCommentsForPost } from './api/forum'; // Fetch post and comments API
```
The import of fetchCommentsForPost throws a typescript warning here, as there is no such function being exported in './api/forum'

YavizGuldalf reviewed (changes requested):
Thanks for the changes. The only problem is that there is a conflict with `main` branch right now, as #451 was recently merged. Can you rebase this branch and force-push so that we can keep a linear git history? Also, can you move the pages related to the forums under a new folder with the path '(tabs)/forum' so that we can have a coherent structure throughout the app. You might need to adjust _layout files. You can see an example of this in the quiz and profile pages in main after merging.

YavizGuldalf reviewed (approved):
Okay, thank you for the changes @anilanlar, it looks good.
